learning_cp/eternity2/features1.jl

```diff
@@ -0,0 +1,198 @@
+#this file contains a few attempts to featurize the Eternity2 problem for learning using FullFeaturedCPNN
+
+struct EternityFeaturization <: SeaPearl.AbstractFeaturization end
+struct EternityFeaturization2 <: SeaPearl.AbstractFeaturization end
+struct EternityFeaturization3 <: SeaPearl.AbstractFeaturization end
+
+function SeaPearl.featurize(sr::SeaPearl.DefaultStateRepresentation{EternityFeaturization,TS}) where TS
+    n = eternity2_generator.n
+    m = eternity2_generator.m
+    k = eternity2_generator.k
+    g = sr.cplayergraph
+    features = zeros(Float32, 17, nv(g))
+    for i in 1:nv(g)
+        cp_vertex = SeaPearl.cpVertexFromIndex(g, i)
+        if isa(cp_vertex, SeaPearl.VariableVertex) && !isa(cp_vertex.variable, SeaPearl.IntVarViewOffset)
+            id = cp_vertex.variable.id
+            #println(id)
+            s = split(id,['_'])[1]
+            l = s[1]
+            t = split(id,['_'])[2]
+            if l=='o'#orientation
+                features[1,i] = 1
+                a,b = parse(Int, split(t,"")[1]), parse(Int, split(t,"")[2])
+                features[2,i] = a
+                features[3,i] = b
+            elseif l=='i'
+                features[8,i] = 1
+                features[9,i] = parse(Int, split(t,"")[1])
+                features[10,i] = parse(Int, split(t,"")[2])
+            else
+                if t[1]=="v" features[11,i] = 1 else features[12,i] = 1 end
+                features[13,i] = parse(Int, t[2])
+                features[14,i] = parse(Int, t[3])
+            end
+        end
+        if isa(cp_vertex, SeaPearl.ConstraintVertex)
+            features[15,i] = 1
+        end
+        if isa(cp_vertex, SeaPearl.ValueVertex)
+            features[16,i] = cp_vertex.value
```

Comment:
Why do we use this kind of encoding instead of a one-hot encoding on values that could maybe carry more information (or make it more understandable by the agent) ? isn't then a possible confusion between rotation and position value ?